src/main/java/de/dhbw/mosbach/inf16a/kochbuch/rezeptservice/RezeptListenController.java

```diff
@@ -0,0 +1,36 @@
+package de.dhbw.mosbach.inf16a.kochbuch.rezeptservice;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * @author Team Chrocorg: Christian Werner, Yoco Harrmann und Georg Frey
+ */
+
+@RestController
+public class RezeptListenController {
+
+    @Autowired
+    private RecipeRepository recipeRepository;
+
+    @RequestMapping("/getRecipesbyTag")
+
+    @GetMapping(value = "/recipes/{tag}")
+    public List<Recipe> getRecipesByTag(String tag) {
+        List hilfsarray = recipeRepository.findAll();    //Alle Rezepte ins Hilfsarray reinspeichern
+        Iterator<Recipe> iter = hilfsarray.iterator();
```

Comment:
Anstelle eines Iterators lässt sich ein anderer Schleifentyp hier sinnvoller verwenden

Reply:
1. Lern erstmal konstruktive Verbesserungsvorschläge zu äußern. Was soll ich denn jetzt ändern? Soll ich eine For-Schleife verwenden? Oder eine For-Each-Schleife? Nur vom Kommentar her kann ich doch nicht raten, was du gerne hättest
2. Ist das für das Projekt hier echt total irrelevant, welche Schleife man da verwendet. Für den Endanwender gibt es da überhaupt keinen Unterschied, was man hier im Backend verwendet

Reply:
...
1. Okay... `for` oder `foreach` aber  bitte nicht manuell den Iterator benutzen, da zu diesem Zweck ja die `foreach` Schleife existiert. 

> Note that this approach does not properly separate the advance operation from the actual data access. 
> aus [Wikipedia](https://en.wikipedia.org/wiki/Iterator#Java)


2. Für das Projekt ist es auch nicht komplett irrelevant wie der Code aussieht . Das ist genauso wie man sagen könnte, dass man das ganze auch auf Assemblerbasis implementieren könnte und es irrelevant sei, wenn es funktioniert, dabei wünsch ich viel Spaß bei der Fehlersuche.
Mach den Code eben nicht komplizierter als er sein muss um sein Zweck zu erfüllen. Dazu zählt, dass man das verwendet, was eine Programmiersprache/Framework einem bietet um "fauler" zu sein.

Reply:
2. Ich habe mich jetzt an das Tutorial gehalten und versucht das zu implementieren. Bei mir läuft jetzt aber die .jar nicht mehr durch mit folgendem Fehler: 

org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'recipeController': Unsatisfied dependency expressed through field 'recipeRepository'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'recipeRepository': Invocation of init method failed; nested exception is org.springframework.data.mapping.PropertyReferenceException: No property tag found for type Recipe! Did you mean 'tags'?

Wo liegt denn der Fehler und was kann ich jetzt machen? Ohne die .jar kann ich halt auch nix testen...